cfme/tests/networks/test_sdn_crud.py

```diff
@@ -41,3 +43,21 @@ def test_sdn_crud(provider, appliance):
     provider.wait_for_delete()
 
     assert not network_provider.exists
+
+
+@pytest.mark.uncollectif(lambda appliance: appliance.version < '5.10',
+                         reason='Delete button was added in CFME 5.10.')
+@pytest.mark.provider([OpenStackProvider], scope='function', override=True)
+def test_router_crud(provider, appliance):
+    router_name = 'test_router_crud-{}'.format(fauxfactory.gen_alphanumeric().lower())
+    routers = appliance.collections.network_routers
+    router = routers.create(router_name, provider, provider.data.networks.cloud_tenant,
+                            provider.data.networks.network_manager)
+    router.delete()
+    provider.refresh_provider_relationships()
+
+    def router_doesnt_exists():
```

Comment:
**Question** Why not just use the NetworkRouter exists method?

Reply:
@jaryn question above still remains. Why not use the existing ```exist``` method?

Reply:
Condensing conversation to L67